armory/utils/metrics.py

```diff
@@ -721,6 +721,47 @@ def video_tracking_mean_iou(y, y_pred):
     return mean_ious
 
 
+def video_tracking_mean_success_rate(y, y_pred):
+    """
+    Mean success rate averaged over all thresholds in {0, 0.05, 0.1, ..., 1.0} and all frames.
+    This function expects to receive a single video's labels/prediction as input.
+
+    y (List[Dict, ...]): list of length equal to number of examples. Each element
+                  is a dict with "boxes" key mapping to (num_frames, 4) numpy array
+    y_pred (List[Dict, ...]): list of length equal to number of examples. Each element
+                  is a dict with "boxes" key mapping to (num_frames, 4) numpy array
+    """
+    _check_video_tracking_input(y, y_pred)
+    if len(y_pred) > 1:
+        raise ValueError(f"y_pred expected to have length of 1, found {len(y_pred)}.")
+
+    thresholds = np.arange(0, 1.05, 0.05)
+    mean_success_rates = (
+        []
+    )  # initialize list that will have length num_videos, which currently is forced to be 1
+    for video_idx in range(len(y_pred)):
+        success = np.zeros(len(thresholds))
+
+        # Selecting first element since y_pred is forced to have length 1
+        y_pred_boxes = y_pred[0]["boxes"]
```

Comment:
Why exactly are you bothering with a loop if y_pred only has predictions for one video?  I would say either remove the loop or change this to `y_pred_boxes = y_pred[video_idx]["boxes"]` if you expect to handle longer y_preds in the future